src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -179,15 +183,15 @@ public function __destruct()
             // Clear local caches when the only remaining handles are about pushed responses
             if (!$this->multi->openHandles) {
                 if ($this->logger) {
-                    foreach ($this->multi->pushedResponses as $url => $response) {
+                    foreach (array_keys($this->multi->pushedResponses) as $url) {
```

Comment:
no real benefit, I'd suggest reverting

Reply:
SA tools will warn about this, not addressing that warning might desensitize people about SA tools warnings.

Reply:
array_keys creates a unless array in memory - I care much more for technical realities that opinions of static analyzers :)

Reply:
I'm not super convinced, because it's an array of strings, and its size is probably dwarfed by the size of the array of values, but your repo, your rules.

Reply:
but this requires a `O(n)` operation to build the array of keys.

Reply:
Right. Moreover, my argument seems invalid at least for phpstan: https://phpstan.org/r/492636a0-0011-4791-9e5e-b92ea9ef6994 There is no warning.

Reply:
> but this requires a `O(n)` operation to build the array of keys.

The loop already is in O(n), so the `array_keys` won't change the complexity class of the operation. But I agree that in a highly optimized code area like this one, we should not create unnecessary temporary arrays. I'll revert this change.